make/autoconf/jdk-options.m4

```diff
@@ -415,19 +415,29 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_ADDRESS_SANITIZER],
       CHECK_AVAILABLE: [
         AC_MSG_CHECKING([if AddressSanitizer (asan) is available])
         if test "x$TOOLCHAIN_TYPE" = "xgcc" ||
-            test "x$TOOLCHAIN_TYPE" = "xclang"; then
+           test "x$TOOLCHAIN_TYPE" = "xclang" ||
+           test "x$TOOLCHAIN_TYPE" = "xmicrosoft"; then
           AC_MSG_RESULT([yes])
         else
           AC_MSG_RESULT([no])
           AVAILABLE=false
         fi
       ],
       IF_ENABLED: [
-        # ASan is simply incompatible with gcc -Wstringop-truncation. See
-        # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85650
-        # It's harmless to be suppressed in clang as well.
-        ASAN_CFLAGS="-fsanitize=address -Wno-stringop-truncation -fno-omit-frame-pointer -fno-common -DADDRESS_SANITIZER"
-        ASAN_LDFLAGS="-fsanitize=address"
+        ASAN_CFLAGS="-fsanitize=address"
```

Comment:
It seems like more of a coincidence than a obvious fact to base the design on that `-fsanitize=address` happens to work on both ms cl.exe and gcc/clang.

Maybe this is the herald of a new age of compiler switches getting cross-compiler standard names (let's hope!), but I think it still makes more sense to not share CFLAGS between microsoft and clang/gcc.

Or, more concretely: There is no need to define `ASAN_CFLAGS` and `ASAN_LDFLAGS` outside the two toolchain specific clauses.

Reply:
Updated to what I think you were requesting. PTAL.

Reply:
Yes, indeed. Thanks!